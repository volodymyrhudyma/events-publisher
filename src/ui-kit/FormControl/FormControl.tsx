import { isEmpty } from 'lodash';
import React, { Children, cloneElement } from 'react';
import { IFormControlProps } from 'src/types';
import Counter from 'src/ui-kit/Counter';
import Error from 'src/ui-kit/Error';
import Label from 'src/ui-kit/Label';
import {
  ContentWrapper,
  ErrorWrapper,
  LabelWrapper,
  RequiredWrapper,
  Wrapper,
} from './styled/FormControl.styled';

const FormControl = ({
  children,
  label,
  required,
  hint,
  error,
  maxLength,
  currentLength,
}: IFormControlProps) => {
  const childrenWithProps = Children.map(children, child =>
    cloneElement(child, { error: !isEmpty(error) })
  );
  return (
    <Wrapper>
      <LabelWrapper>
        <Label color={error ? 'error' : 'primary'}>{label}</Label>
        {required && <RequiredWrapper>*</RequiredWrapper>}
      </LabelWrapper>
      <ContentWrapper>
        {childrenWithProps}
        {hint}
        {maxLength && (
          <Counter maxLength={maxLength} currentLength={currentLength} />
        )}
      </ContentWrapper>
      <ErrorWrapper>{error && <Error>{error}</Error>}</ErrorWrapper>
    </Wrapper>
  );
};

FormControl.defaultProps = {
  required: false,
};

export default FormControl;
