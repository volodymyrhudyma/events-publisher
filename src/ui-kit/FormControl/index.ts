import Component from './FormControl';

export default Component;
