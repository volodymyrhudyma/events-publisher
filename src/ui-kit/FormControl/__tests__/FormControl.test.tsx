import React from 'react';
import { shallow } from 'enzyme';
import FormControl from '../FormControl';

describe('FormControl component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <FormControl label="Label">FormControl</FormControl>
    );
    expect(component).toMatchSnapshot();
  });
});
