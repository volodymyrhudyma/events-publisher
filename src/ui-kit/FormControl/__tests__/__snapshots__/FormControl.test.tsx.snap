// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControl component should render correctly 1`] = `
<styled.div>
  <styled.div>
    <Label
      color="primary"
      uppercase={true}
    >
      Label
    </Label>
  </styled.div>
  <styled.div>
    <Component
      error={false}
      key=".0"
    />
  </styled.div>
  <styled.div />
</styled.div>
`;
